keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -148,3 +148,173 @@ def get_config(self):
             }
         )
         return config
+
+    def _augment(self, inputs):
+        image = inputs.get("images", None)
+        label = inputs.get("labels", None)
+        bounding_box = inputs.get("bounding_boxes", None)
+        transformation = self.get_random_transformation(
+            image=image, label=label, bounding_box=bounding_box
+        )  # pylint: disable=assignment-from-none
+        image = self.augment_image(image, transformation=transformation)
+        result = {"images": image}
+        if label is not None:
+            label = self.augment_label(label, transformation=transformation)
+            result["labels"] = label
+        if bounding_box is not None:
+            # changed augment_bounding_box requires image as well
+            # for unnormalizing bbox coordinates
+            bounding_box = self.augment_bounding_box(
+                image, bounding_box, transformation=transformation
+            )
+            result["bounding_boxes"] = bounding_box
+        return result
+
+    def augment_bounding_box(self, image, bounding_boxes, transformation):
+        """args: image : takes a single image H,W,C,
+        bounding_boxes: take bbox coordinates [N,4] -> [y1,x1,y2,x2]
+        transformation: takes tuple for x,y transformation None if no transformation"""
+        x, y = transformation
+        if x is not None:
+            bounding_boxes = self.augment_horizontal(image, bounding_boxes, x)
+        if y is not None:
+            bounding_boxes = self.augment_vertical(image, bounding_boxes, y)
+        return tf.clip_by_value(
```

Comment:
bounding boxes are not guaranteed. to be between [0, 1].  They are in PascalVOC format, but in COCO they are absolute coordinates.  please update the clip to rely on `tf.shape(image)`.  You will need to clip differently on the width and height axes.

Reply:
@LukeWood So the idea is return the transformed shear bboxes as the same input format as they were received?
For ex: yolo, voc, coco and returned back in same format?

Reply:
@divyashreepathihalli Can you help?
Blocked on this.

Reply:
Hey @pranavjadhav001, we are still doing some design work around bounding box handling.  The solution to this problem is not clear.

For now, you can always return the bounding boxes in corners format.